docs/dev.rst

```diff
@@ -1,20 +1,570 @@
+
+
+
+Reference a Specific Tag
+************************
+
+To reference a version other than HEAD and the latest production release, use the ``tag`` option to specify a tag from the target repository. This option is most useful for testing against beta versions of underlying packages, or recreating specific org environments for debugging.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+              tag: beta/1.47-Beta_2
+
+..
+
+    The EDA repository's tag, ``beta/1.47-Beta_2``, is used instead of the latest production release of EDA (1.46, for this example). This tag lets a build environment use features in the next production release of EDA that are already merged but not yet included in a production release.
+
+
+
+Skip ``unpackaged/*`` in Reference Repositories
+***********************************************
+
+If the repository you are referring to has dependency metadata under unpackaged/pre or unpackaged/post, use the ``skip`` option to skip deploying that metadata with the dependency.
+
+    Example: Salesforce EDA
+
+.. code-block:: yaml
+
+    project:
+        dependencies:
+            - github: https://github.com/SalesforceFoundation/EDA
+              skip: unpackaged/post/course_connection_record_types
+
+
+Automatic Cleaning of ``meta.xml`` Files on Deploy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To let CumulusCI fully manage the project's dependencies, the ``deploy`` task (and other tasks based on ``cumulusci.tasks.salesforce.Deploy``, or subclasses of it) automatically removes the ``<packageVersion>`` and its children from all ``meta.xml`` files in the deployed metadata. Removing these elements does not affect the files on the filesystem.
+
+The reason for stripping ``<packageVersion>`` elements on deploy is that the target Salesforce org automatically adds them back with the installed version of the referenced namespace. This feature lets CumulusCI fully manage dependencies, and avoids rushing a new commit of ``meta.xml`` files when a new underlying package version is available.
+
+.. note:: If the metadata being deployed references namespaced metadata that does not exist in the currently installed package, the deployment throws an error as expected.
+
+The automatic cleaning of ``meta.xml`` files can be disabled by setting the ``clean_meta_xml`` task option to ``False``. Prior to the addition of this functionality, there were unnecessary delays in the CumulusCI release cycle due to the need to create a new commit on ``main`` (and thus a feature branch, PR, code review, and so on) just to update the ``meta.xml`` files. CumulusCI's GitHub dependency functionality already handles requiring a new production release, so the only reason to do this commit was for the ``meta.xml`` files. Automatically cleaning the meta.xml files on deploy eliminates the need for this commit.
```

Comment:
I think we're going into too much depth and detail here. This feature should be made known to users, but it's almost always behind the scenes.